DQM/HcalTasks/plugins/QIE11Task.cc

```diff
@@ -144,42 +186,88 @@ QIE11Task::QIE11Task(edm::ParameterSet const& ps):
 		QIE11DataFrame frame = static_cast<QIE11DataFrame>((*cqie11)[i]);
 		DetId did = frame.detid();
 		HcalElectronicsId eid = HcalElectronicsId(_ehashmap.lookup(did));
+		if (!eid.rawId()) {
+		  continue;
+		}
 		int fakecrate = -1;
 		if (eid.crateId() == 34) fakecrate = 0;
 		int index = fakecrate * 12 + (eid.slot() - 10) - 1;
 
 		//	compute the signal, ped subracted
+		CaloSamples digi_fC = hcaldqm::utilities::loadADC2fCDB<QIE11DataFrame>(_dbService, did, frame);
 //		double q = hcaldqm::utilities::aveTS_v10<QIE11DataFrame>(frame,
 //			constants::adc2fC[_ped], 0, frame.samples()-1);
 
 		//	iterate thru all TS and fill
 		for (int j=0; j<frame.samples(); j++)
 		{
+			double q_pedsub = hcaldqm::utilities::adc2fCDBMinusPedestal<QIE11DataFrame>(_dbService, digi_fC, did, frame, j);
 			if (index == 0 || index == 1) {
 				//	shapes are after the cut
-				_cShapeCut_EChannel[index].fill(eid, j, adc2fC[frame[j].adc()]);	
-				_cTDCvsTS_EChannel[index].fill(eid, j, frame[j].tdc());	
+				_cShapeCut_EChannel[index].fill(eid, j, q_pedsub);	
+				_cLETDCvsTS_EChannel[index].fill(eid, j, frame[j].tdc());	
 
 				//	w/o a cut
-				_cTDCvsADC_EChannel[j][index].fill(eid, frame[j].adc(), 
+				_cLETDCvsADC_EChannel[j][index].fill(eid, frame[j].adc(), 
 					frame[j].tdc());
-				_cTDC_EChannel[j][index].fill(eid, frame[j].tdc());
+				_cLETDC_EChannel[j][index].fill(eid, frame[j].tdc());
 				if (frame[j].tdc() < 50) {
 					// Each TDC count is 0.5 ns. 
 					// tdc == 62 or 63 means value was below or above threshold for whole time slice. 
-					_cTDCTime_EChannel[index].fill(eid, j*25. + (frame[j].tdc() / 2.));
+					_cLETDCTime_EChannel[index].fill(eid, j*25. + (frame[j].tdc() / 2.));
 				}
 				_cADC_EChannel[j][index].fill(eid, frame[j].adc());
 			}
-			_cShapeCut.fill(eid, j, adc2fC[frame[j].adc()]);
+			_cShapeCut.fill(eid, j, q_pedsub);
 
-			_cTDCvsADC.fill(frame[j].adc(), frame[j].tdc());
-
-			_cTDC.fill(eid, frame[j].tdc());
+			_cLETDCvsADC.fill(frame[j].adc(), frame[j].tdc());
+			if (frame[j].tdc() < 50) {
+				_cLETDCTimevsADC.fill(frame[j].adc(), j*25. + (frame[j].tdc() / 2.));
+			}
+			_cLETDC.fill(eid, frame[j].tdc());
 
 			_cADC.fill(eid, frame[j].adc());
 
 		}
+
+		// Timing channels for phase scan
+		for (int iChan = 0; iChan < 4; ++iChan) {
+			if (!_filter_timingChannels[iChan].filter(HcalDetId(did))) {
+				// For local runs, hack something in for lumisection
+				int ls = _currentLS;
+				if (_ptype == fLocal) {
+					ls = (int)(e.id().event() / 20);
```

Comment:
This was an attempt to "fake" a LS number for local runs, for two of the histograms with LS on the x-axis. The histograms are not needed for local runs, so I've removed this /20, and skip the histograms for local runs.